src/public/parts/shells/code/public/src/templates/ai_source_panel.html

```
@@ -0,0 +1,9 @@
+<div class="modal-box">
+	<h2 class="text-lg font-bold" data-i18n="code.aiSource.panelTitle"></h2>
+	<div id="ai-source-list" class="py-2"></div>
+	<div class="modal-action">
+		<a class="btn btn-sm" href="/parts/shells:serviceSourceManage/" target="_blank" rel="noopener" data-i18n="code.aiSource.manage"></a>
+		<form method="dialog"><button class="btn btn-sm" data-i18n="util.common.close"></button></form>
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)